pkg/asset/agent/joiner/clusterinfo.go

```diff
@@ -2,24 +2,44 @@ package joiner
 
 import (
 	"context"
-	"net/url"
+	"fmt"
 
+	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/yaml"
 
+	hiveext "github.com/openshift/assisted-service/api/hiveextension/v1beta1"
 	configclient "github.com/openshift/client-go/config/clientset/versioned"
 	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/agent"
 	"github.com/openshift/installer/pkg/asset/agent/workflow"
+	"github.com/openshift/installer/pkg/types"
 )
 
 // ClusterInfo it's an asset used to retrieve config info
-// from an already existing cluster.
+// from an already existing cluster. A number of different resources
+// are inspected to extract the required configuration.
 type ClusterInfo struct {
-	ClusterID  string
-	APIDNSName string
-	PullSecret string
+	Client          kubernetes.Interface
+	OpenshiftClient configclient.Interface
+
+	ClusterID                     string
+	ClusterName                   string
+	Version                       string
+	ReleaseImage                  string
+	APIDNSName                    string
+	PullSecret                    string
+	Namespace                     string
+	UserCaBundle                  string
+	Proxy                         *types.Proxy
+	Architecture                  string
+	ImageDigestSources            []types.ImageDigestSource
+	DeprecatedImageContentSources []types.ImageContentSource
+	PlatformType                  hiveext.PlatformType
+	SSHKey                        string
```

Comment:
Do you need to add InstallConfig fields like `CPUPartitioning`,`FIPS`, or any of the `Networking` fields?

Reply:
That's a good question @bfournie. @zaneb @rwsu @pawanpinjarkar what are your thoughts about it? I'd say that in this first iteration we could try to address just the min fields required, and eventually add them later in successive iterations

Reply:
OK, with the caveat that we can't get beyond Dev Preview without this stuff.

Reply:
As a first iteration I meant the current PR. Since there are others still pending based on this one (#8080 and #8093), and others will follow as part the owner story [AGENT-850](https://issues.redhat.com/browse/AGENT-850), my preference would be to land the current PR as it is and address such points in successive PRs